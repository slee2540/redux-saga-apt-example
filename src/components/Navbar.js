import React, { Component } from "react";

class Navbar extends Component {
  constructor() {
    super();
    this.state = {
      count: 0
    };
  }

  makeIncrementer = amount => {
    const { count } = this.state;
    this.setState({
      count: count + amount
    });
  };

  // increment = this.makeIncrementer(1);
  // decrement = this.makeIncrementer(-1);

  increment = () => this.makeIncrementer(1);

  decrement = () => this.makeIncrementer(-1);

  render() {
    const { increment, decrement } = this;
    const { count } = this.state;
    return (
      <div>
        <p>Count: {count}</p>
        <button type="button" className="increment" onClick={increment}>
          Increment count
        </button>
        <button type="button" className="decrement" onClick={decrement}>
          Decrement count
        </button>
      </div>
    );
  }
}

export default Navbar;

// export default class Navbar extends Component {
//   constructor(props) {
//     super(props);

//     this.state = {
//       expand: "md",
//       position: null,
//       bgColor: "rgb(240,240,240)",
//       textColor: "rbg(100,100,100)",
//       isOpen: false
//     };
//   }

//   render() {
//     return (
//       <div id="Navbar">
//         <p>테스트입니다.</p>
//       </div>
//     );
//   }
// }
