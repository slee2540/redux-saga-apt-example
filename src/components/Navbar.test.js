import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "../setupTests";
import Navbar from "./Navbar";

// configure({ adapter: new Adapter() });

describe("Navbar component", () => {
  it("starts with a count of 0", () => {
    const wrapper = shallow(<Navbar />);
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: 0");
  });

  it("increments count by 1 when the increment button is clicked", () => {
    const wrapper = shallow(<Navbar />);
    const incrementBtn = wrapper.find("button.increment");
    incrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: 1");
  });

  it("decrements count by 1 when the decrement button is clicked", () => {
    const wrapper = shallow(<Navbar />);
    const decrementBtn = wrapper.find("button.decrement");
    decrementBtn.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual("Count: -1");
  });

  it("matches the snapshot", () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
