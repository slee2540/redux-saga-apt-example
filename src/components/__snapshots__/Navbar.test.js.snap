// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
    type="button"
  >
    Increment count
  </button>
  <button
    className="decrement"
    onClick={[Function]}
    type="button"
  >
    Decrement count
  </button>
</div>
`;
